### PR TITLE
Add docs for Java Maven plugin

### PR DESCRIPTION
**Pull request by lbloder**
Adds documentation for the Java Maven Plugin that allow auto-install of Sentry Dependencies and Source-Bundle upload.

## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

*Describe your changes here. If your PR relates to or resolves an issue, add a link to that too.*

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/platform-includes/source-context/java.mdx [hunk 1]
```diff
@@ -245,15 +245,18 @@ Add the Sentry Maven Plugin to your project by adding the following lines to you
...
+            <!-- Required to allow auto-install of Sentry SDK and Integrations -->
+            <extensions>true</extensions>
             <configuration>
                 <!-- for showing output of sentry-cli -->
                 <debugSentryCli>true</debugSentryCli>
 
-                <!-- download the latest sentry-cli and provide path to it here -->
+                <!-- optionally specify the path to sentry-cli -->
                 <!-- download it here: https://github.com/getsentry/sentry-cli/releases -->
                 <!-- minimum required version is 2.17.3 -->
```
Comment (adinauer) [inline src/platform-includes/source-context/java.mdx:257]:
```suggestion
                <!-- minimum required version is 2.21.2 -->
```

`m` Probably safer to suggest the version that supports org auth tokens for self hosted.

File: src/platforms/java/common/maven/index.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    </plugins>
+    ...
+  </build>
+...
+</project>
+```
+
+### Configure
+
+We expose the following configuration values for the plugin in `pom.xml`:
```
Comment (adinauer) [inline src/platforms/java/common/maven/index.mdx:40]: `l` Maybe explain here that this sample adds a `<configuration>` element to the `pom.xml` sample above?

File: src/platforms/java/common/maven/index.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+```xml
+<project>
+  ...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.sentry</groupId>
+        <artifactId>sentry-maven-plugin</artifactId>
+        <version>{{@inject packages.version('sentry.java.maven-plugin', '0.0.2') }}</version>
+        <extensions>true</extensions>
```
Comment (adinauer) [inline src/platforms/java/common/maven/index.mdx:28]:
```suggestion
        <!-- Required to allow auto-install of Sentry SDK and Integrations -->
        <extensions>true</extensions>
```

`l`

**Review by shanamatthews — APPROVED** [review]
LGTM!

**Pull request merged**